app/models/school_time.rb

```diff
@@ -97,33 +94,79 @@ def to_analytics
 
   private
 
+  #Check whether this SchoolTime overlaps with other SchoolTimes associated with
+  #the same school. This doesn't query the database, because we also need to do
+  #this validation when adding multiple times to a school as part of a form update.
+  #When rails does this is runs the validation for all models, then inserts them
+  #so doing database queries for the time ranges was allowing invalid data to be
+  #inserted
+  def overlapping(usage_type)
+    day = overlapping_days
+    calendar_period = overlapping_calendar_periods
+    overlapping = false
+    school.school_times.each do |other|
+      if other != self &&
+         usage_type == other.usage_type &&
+         day.include?(other.day) &&
+         calendar_period.include?(other.calendar_period) &&
+         overlapping_times?(other)
+        overlapping = true
+      end
```

Comment:
small nitpick but wondering if this might be easier to read as
```
overlapping = true if other != self &&
                      usage_type == other.usage_type &&
                      day.include?(other.day) &&
                      calendar_period.include?(other.calendar_period) &&
                      overlapping_times?(other)
 ```

Reply:
maybe also break early once overlapping is set true?

Reply:
Have updated, can you have another look?

Reply:
:+1: looks good to me.